src/main/java/com/redhat/devtools/lsp4ij/settings/jsonSchema/LSPServerConfigurationJsonSchemaManager.java

```diff
@@ -0,0 +1,50 @@
+package com.redhat.devtools.lsp4ij.settings.jsonSchema;
+
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LSPServerConfigurationJsonSchemaManager {
+
+    public static LSPServerConfigurationJsonSchemaManager getInstance(@NotNull Project project) {
+        return project.getService(LSPServerConfigurationJsonSchemaManager.class);
+    }
+
+    private final List<LSPServerConfigurationJsonSchemaFileProvider> providers;
+
+    public LSPServerConfigurationJsonSchemaManager(@NotNull Project project) {
+        providers = new ArrayList<>();
+        for (int i = 0; i < 100; i++) {
+            providers.add(new LSPServerConfigurationJsonSchemaFileProvider(i, project));
+        }
+    }
+
+    public List<LSPServerConfigurationJsonSchemaFileProvider> getProviders() {
+        return providers;
+    }
+
+    @Nullable
+    public Integer getUnusedIndex() {

```

Comment:
Okay, I think I see what's happening here. You're trying to have the index be part of the unique name/key, and you're also trying to keep the range of available index values contiguous, so you're (re)allocating index values as-needed.

Is there a reason to do it this way vs. using something like a UUID for uniqueness?

Reply:
The reason is that I have list of JsonSchemaFileProvider and  the manager returns the index of JsonSchemaFileProvider which is marked as unused. I use index as key and no need to manage a map for that (to use UUID)